service/resource_router.go

```diff
@@ -0,0 +1,33 @@
+package service
+
+import (
+	"context"
+
+	"github.com/giantswarm/kvm-operator/service/keyv2"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+// newResourceRouter determines which resources are enabled based upon the
+// version in the version bundle.
+func newResourceRouter(resources []framework.Resource) func(ctx context.Context, obj interface{}) ([]framework.Resource, error) {
+	return func(ctx context.Context, obj interface{}) ([]framework.Resource, error) {
+		var enabledResources []framework.Resource
+
+		customObject, err := keyv2.ToCustomObject(obj)
+		if err != nil {
+			return enabledResources, microerror.Mask(err)
```

Comment:
Please return the zero value in case an error is returned. The zero value of a slice is `nil`. This code has potential to do weird things as soon as the list got modified on its way somehow. Such things are quite easy to miss.